storage/bucket_finder.go

```diff
@@ -0,0 +1,62 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb/v2"
+)
+
+var _ BucketFinder = (*AllBucketsFinder)(nil)
+
+// AllBucketsFinder is responsible for iterating over the paginated reponses from FindBuckets to return every bucket
+type AllBucketsFinder struct {
+	bucketSvc influxdb.BucketService
+}
+
+func NewAllBucketsFinder(bucketSvc influxdb.BucketService) *AllBucketsFinder {
```

Comment:
In this PR, I see this new constructor, and one use of it in a test. Is there follow-on work, or do you mean for the contents of this file to live in `bucket_finder_test.go`?

Reply:
There is a follow up PR and I also added the constructor to cmd/launcher.go